tokio/src/macros/select.rs

```diff
@@ -0,0 +1,788 @@
+/// Wait on multiple concurrent branches, returning when the **first** branch
+/// completes, cancelling the remaining branches.
+///
+/// The `select` macro accepts one or more branches with the following pattern:
+///
+/// ```text
+/// <pattern> = <async expression> (, if <condition>)? => <handler>,
+/// ```
+///
+/// Additionally, the `select!` macro may include a single, optional `else`
+/// branch, which evaluates if none of the other branches match their patterns:
+///
+/// ```text
+/// else <expression>
+/// ```
+///
+/// The macro aggregates all `<async expression>` expressions and runs them
+/// concurrently on the **current** task. Once the **first** expression
+/// completes with a value that matches its `<pattern>`, the `select!` macro
+/// returns the result of evaluating the completed branch's `<handler>`
+/// expression.
+///
+/// Additionally, each branch may include an optional `if` condition. This
+/// condition is evaluated **before** the <async expression>. If the condition
+/// returns `false`, the branch is entirely disabled. This capability is useful
+/// when using `select!` within a loop.
+///
+/// The complete lifecycle of a `select!` expression is as follows:
+///
+/// 1. Aggregate the `<async expression>`s from each branch.
+/// 2. Evaluate all provded `<condition>` expressions. If the condition returns
+///    `false`, disable the branch.
+/// 3. Concurrently await on the results for all remaining `<async expression>s.
+/// 4. Once an `<async expression>` returns a value, attempt to apply the value
+///    to the provided `<pattern>`, if the pattern matches, evaluate `<handler>`
+///    and return. If the pattern **does not** match, disable the current branch
```

Comment:
`disable` sounds a bit like it is permanently disabled - but it's actually just not used for the current iteration - right? Maybe use another phrase:
- skipped
- ignored
- not taken into account

Reply:
It is for multiple iterations, but internal to that `select!`. I could specify "disable the branch for the remainder of the call to `select!`?

Reply:
Yes that also sounds good to me

Reply:
Or „disables polling of the branch for the remainder of the call to select“

Reply:
I updated to "disable for the remainder of the current call to select!`